students/eunhyung/users/views.py

```diff
@@ -1,3 +1,37 @@
-from django.shortcuts import render
+import json
+import re
 
-# Create your views here.
+from django.http import JsonResponse
+from django.views import View
+
+from users.models import User
+
+class SignUpView(View):
+    def post(self, request):
+        
+        data = json.loads(request.body)
+        
+        valid_email    = re.compile('^[a-zA-Z0-9]+@[a-zA-Z]+\.[a-zA-Z]+$')
+        valid_password = re.compile('^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,}$')
+        
+        if data['email'] == '' or data['password'] == '':
+            return JsonResponse({"message": "KEY_ERROR"}, status=400)
+
+        if not valid_email.match(data['email']):
+            return JsonResponse({'message':'Email is invalid.'}, status=400)
+         
+        if not valid_password.match(data['password']):
+            return JsonResponse({'message':'Password is invalid.'}, status=400)
```

Comment:
1. 정규표현식을 사용할 경우, 자주 사용되는 경우가 아니라면 `compile()`해서 사용하지 않고, `match()`를 사용하는 것을 공식 문서에서 추천하고 있습니다.
2. 정규 표현식 같은 경우 변수라기 보다는 상수입니다. python에서 상수는 대문자로 만듭니다.

```suggestion
        REGEX_EMAIL    = '^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$'
        REGEX_PASSWORD = '^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,}$'

        if not re.match(REGEX_EMAIL, data['email'):
            return JsonResponse({'message':'Email is invalid.'}, status=400)
         
        if not re.match(REGEX_PASSWORD, data['password'):
            return JsonResponse({'message':'Password is invalid.'}, status=400)
```

Reply:
넵